vendor/blockchain_contracts/tests/rfc003_bitcoin_htlc.rs

```diff
@@ -1,53 +1,58 @@
 #![warn(unused_extern_crates, missing_debug_implementations, rust_2018_idioms)]
 #![deny(unsafe_code)]
 
-mod htlc_harness;
+pub mod ethereum_wallet;
+pub mod htlc_harness;
+pub mod parity_client;
 
-use crate::htlc_harness::{sleep_until, CustomSizeSecret};
+use crate::htlc_harness::{CustomSizeSecret, Timestamp, SECRET, SECRET_HASH};
 use bitcoin_rpc_test_helpers::RegtestHelperClient;
 use bitcoin_support::{
     serialize_hex, Address, BitcoinQuantity, Network, OutPoint, PrivateKey, PubkeyHash,
     TransactionId,
 };
-use bitcoin_witness::{
-    PrimedInput, PrimedTransaction, UnlockParameters, Witness, SEQUENCE_ALLOW_NTIMELOCK_NO_RBF,
-};
+use bitcoin_witness::{PrimedInput, PrimedTransaction, UnlockParameters, Witness};
 use bitcoincore_rpc::RpcApi;
-use comit_node::swap_protocols::{
-    rfc003::{bitcoin::Htlc, Secret, SecretHash},
-    Timestamp,
-};
+use blockchain_contracts::bitcoin::rfc003::bitcoin_htlc::BitcoinHtlc;
 use secp256k1_support::KeyPair;
 use spectral::prelude::*;
-use std::str::FromStr;
+use std::{str::FromStr, thread::sleep, time::Duration};
 use testcontainers::{clients::Cli, images::coblox_bitcoincore::BitcoinCore, Docker};
 
 impl CustomSizeSecret {
-    pub fn unlock_with_secret(self, htlc: &Htlc, keypair: KeyPair) -> UnlockParameters {
-        let public_key = keypair.public_key();
+    pub fn unlock_with_secret(self, htlc: &BitcoinHtlc, keypair: KeyPair) -> UnlockParameters {
+        let UnlockParameters {
+            mut witness,
+            sequence,
+            locktime,
+            prev_script,
+        } = htlc.unlock_with_secret(keypair, [0u8; 32]);
+        let secret = self.0;
+
+        for w in &mut witness {
```

Comment:
❓I'm not sure if I can follow what's happening here.

Reply:
The motivation was to _not_ duplicate the code from `unlick_with_secret` (which was done before) but rather replace the secret on the witness stack with a custom one (note that this is the `CustomSizeSecret` type).

Hence I have to iterate through the witness stack, find the `Data` and replace it with the secret that I actually want to put there.

I'll add a comment that explains this :sweat_smile: